internal/wasm/interpreter/interpreter_test.go

```diff
@@ -89,15 +89,13 @@ func TestEngine_Call_HostFn(t *testing.T) {
 	module := &wasm.ModuleInstance{MemoryInstance: memory}
 	modCtx := wasm.NewModuleContext(context.Background(), e, module)
 	f := &wasm.FunctionInstance{
-		HostFunction: &hostFn,
-		FunctionKind: wasm.FunctionKindGoModuleContext,
-		FunctionType: &wasm.TypeInstance{
-			Type: &wasm.FunctionType{
-				Params:  []wasm.ValueType{wasm.ValueTypeI64},
-				Results: []wasm.ValueType{wasm.ValueTypeI64},
-			},
+		GoFunc: &hostFn,
+		Kind:   wasm.FunctionKindGoModuleContext,
+		Type: &wasm.FunctionType{
+			Params:  []wasm.ValueType{wasm.ValueTypeI64},
+			Results: []wasm.ValueType{wasm.ValueTypeI64},
 		},
-		ModuleInstance: module,
```

Comment:
the main reason we use the type name for the field name is when there's a method of the same name (so can't), or the name is the same as a reserved word.

I reverted the longer names in Function to improve readability. There are other places we can shorted also.